Metadata Schema Charter

lauraclarke opened the pull request:
~**October 15**: Last call for community reviews~
**October 23**: Last call for oversight reviews
**October 25**: Approval

lauraclarke committed:
This is the start of the metadata schema branch

lauraclarke committed:
First draft for metadata schema charter

lauraclarke committed:
Fixing header

lauraclarke committed:
Fixing another header

lauraclarke committed:
Fixing a link

lauraclarke committed:
Fixing another header

tburdett commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+The HCA Metadata schema committee creates and maintains the Human Cell Atlas metadata schema in JSON format. These schema are designed to capture and provide structure for the descriptive scientific metadata associated with HCA datasets. These schemas aim to ensure the FAIRness* of the HCA data [The FAIR Guiding Principles for scientific data management and stewardship](https://www.nature.com/art ...
+
+## Objective
```
I'd like to see a definition of how comprehensive metadata schemas are expected to be in order to deliver on scientific aims. I think it is implicit here that schemas will aim to represent the contents of data submitted to the HCA - and it is worth making this explicit - but I would also like to see a declaration about whether the metadata team aim to produce a minimum information style standard, a fully comprehensive one, or something in between (and if something in between, what?)

lauraclarke replied:
I have added another paragraph to the description which addresses this. Do let me know what you think.

tburdett commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+2. Ensure that HCA metadata supports reproducibility and reusability.
+3. Ensure that HCA metadata has a consistent representation and is computationally queryable.
+4. Define JSON schema which enable objectives 1 and 2.
+5. Specify the entity structure for HCA metadata schema.
```
I think this is what I would consider to be the HCA core domain model (and have used this terminology in the infrastructure charter which I am writing). If so, this is something that needs to be carefully negotiated across components as software has dependencies on this. I would suggest making it explicit that this is a negotiation, and we should review for consistency once I have published the PR for ingest infrastructure

lauraclarke replied:
Can you propose a phrasing for the negotiation as I think 2, 3 and 4 are clearly in the metadata domain but 4 seems to be the thing which is more grey as to who is responsible for is?

lauraclarke replied:
I have edited the line about entity models to try and reflect this.

lauraclarke replied:
@tburdett can you confirm you are happy with the edited phrasing

tburdett commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## In-scope
+
+* Defining the specification for the schema and the the schema itself.
```
> the the schema itself

lauraclarke replied:
@tburdett can you look again and see if you like the improved phrasing

tburdett commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+* Incorporate community feedback to extend and improve the schema.
+* Ensuring the schema is compatible with open source software packages supporting the same draft of the JSON schema.
+* Provide a test framework to enable community members to test if their proposed extensions are functional and meet our style guides.
+* Providing exemplar data in the schema format.
```
I think you mean exemplar data in JSON format that validates against the schema?

tburdett commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+## Out-of-scope
+
+* Building interfaces to translate metadata between HCA metadata JSON and other serialization formats
+* Building services to to index or query the metadata
```
> to to index or query

tburdett reviewed (commented):
There's a general overlap (and some tension) here with the Query Service Charter PR. I understand that the metadata team is trying to enable introspection to make handling evolution of schemas easy to handle with introspection, whereas the query service team seem to prefer an ETL-based approach (presumably to align to some consumer-centric model). Aligning approaches would seem to be pretty urgent, otherwise the two teams will cause a big overhead working against each other. @mweiden do you have a view here? I am happy to jump on a call if I can be helpful

brianraymor committed:
Update charter.md

brianraymor committed:
Update charter.md

brianraymor commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+## Description
+
+The HCA Metadata schema committee creates and maintains the Human Cell Atlas metadata schema in JSON format. These schema are designed to capture and provide structure for the descriptive scientific metadata associated with HCA datasets. These schemas aim to ensure the FAIRness* of the HCA data [The FAIR Guiding Principles for scientific data management and stewardship](https://www.nature.com/art ...
+
```
Perhaps some compression:

`... schema in JSON format. These schemas capture and define structure for the descriptive scientific metadata associated with HCA datasets and ensure the [FAIRness](https://www.nature.com/articles/sdata201618) of the HCA data.`

brianraymor commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+[Laura Clarke](mailto:laura@ebi.ac.uk)
+### Product Owner
+[Norman Morrison](mailto:norman@ebi.ac.uk)
+### Technical Lead
```
If there is no **Technical Lead** role, then this section should be omitted or include a comment like _Intentionally left blank_. I'm open to whatever the DCP community prefers.

brianraymor committed:
Update charter.md

brianraymor commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+### Scientific "guardrails" 
+
+In order for the HCA metadata schema to meet the needs of the community, the defined schema must reflect the types of data which are being generated in order to build the Human Cell Atlas. The HCA Metadata schema committee will seek guidance from the scientific leadership of the HCA to help understand and prioritize new data types for addition to the schema collection as well as collecting feedba ...
```
Can you share more specifics about the _scientific leadership_ (is this the Governance Group) and the process for seeking guidance? This can help define charters for Science Governance based on DCP requirements.

@jonahcool - Any suggestions?

brianraymor commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## Out-of-scope
+
+* Building interfaces to translate metadata between HCA metadata JSON and other serialization formats
```
_Building interfaces to translate HCA metadata from JSON to other serialization formats_

brianraymor committed:
Update charter.md

mweiden commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## Description
+
+The HCA Metadata schema committee creates and maintains the Human Cell Atlas metadata schema in JSON format. These schema are designed to capture and provide structure for the descriptive scientific metadata associated with HCA datasets. These schemas aim to ensure the FAIRness* of the HCA data [The FAIR Guiding Principles for scientific data management and stewardship](https://www.nature.com/art ...
```
* nit: Define the acronyms upon first usage so this document is more accessible: `The Human Cell Atlas (HCA) ...`
* Fix capitalization of the `Metadata Schema Committee`
* nit: I'd just link FAIR as `... [FAIRness](https://www.nature.com/articles/sdata201618) of the HCA data.`

lauraclarke replied:
Thanks for the suggestions, updates made.

mweiden commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+4. Define JSON schema which enable objectives 1 and 2.
+5. Specify the entity structure for HCA metadata schema.
+6. Define style guides and release process.
+7. Ensure the defined metadata schema provides a positive user experience for those who read and write them.
```
nit: These are not full sentences. I would omit the period at the end of each.

lauraclarke replied:
Thanks, updates made

mweiden commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+* Providing exemplar data in the schema format.
+* Working with data contributors to understand what types of descriptive fields they can supply for their samples and experiments.
+* Working with data consumers to understand what descriptive fields they need in order to be able to analyse and visualise the HCA data.
+* Extending the schema to meet the needs of data contributors and data consumers based on the feedback they provide.
```
* nit: These are not full sentences. I would omit the period at the end of each.
* nit: Some of these bullets are written in active voice and some in passive voice. Choose one, preferably active voice, and use it consistently throughout the document.

lauraclarke replied:
Hopefully should now be consistent

mweiden commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## Milestones
+
+During 2018/2019 The metadata charter committee will work to:
```
* Fix capitalization
* nit: I'd remove the `:` and make this a full sentence. Something like `... will work towards the following milestones.`

lauraclarke replied:
Thanks, hopefully now fixed.

mweiden commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+### Product Owner
+[Norman Morrison](mailto:norman@ebi.ac.uk)
+### Technical Lead
+
```
Will there be a technical lead? Delete this section if not.

lauraclarke replied:
There will be, we are just having some internal discussions about who it will be before I add a name

mweiden commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+# Metadata Schema
```
nit: If this document describes `Metadata Schema Committee`, then that should probably be the title and directory name.

lauraclarke replied:
We probably need to have a broader discussion about style of how we refer to the charters etc in these documents so we can all be consistent across the different documents but for now I have removed the word committee.

mweiden reviewed (commented):
Just a few stylistic comments. Please pardon that they are mostly nit-picks.

Commit:
Update based on feedback from @mweiden and @diekhans

lauraclarke committed:
Adding Mark Diekhans as tech lead

TimothyTickle commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,66 @@
+
+## Description
+
+The Human Cell Atlas (HCA) Metadata Schema are JSON format schema. These schema are designed to capture and provide structure for the descriptive scientific metadata associated with HCA datasets. These schemas aim to ensure [the FAIRness of the HCA data](https://www.nature.com/articles/sdata201618).
```
Although JSON is currently used for the metadata and I am in no way asking for changes, I wonder if JSON should be specifically named in the charter. It seems too much implementation detail in the description and locks you into that format (if you ever decide to update).

lauraclarke replied:
As JSON is so fundamental to how the schemas work I think it needs to be in the charter. If we were to decide to move to YAML or protobuff it would need rechartering so not mentioning JSON seems like needless obfuscation

TimothyTickle commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,66 @@
+1. Ensuring that HCA data are findable and the schemas are computationally queryable
+2. Ensuring that HCA metadata supports reproducibility and reusability
+3. Ensuring that HCA metadata has a consistent representation and can accurately represent the scientific description of a dataset in a computational readable manner
+4. Defining JSON schema which enable objectives 1 and 2
```
This use of the word JSON also seems unnecessary and more generic "Defining schema which..." may be more appropriate.

diekhans replied:
agreed, JSON is an implementation. not an objective.

malloryfreeberg replied:
I'm a little confused about where the line is for including/excluding implementation choices in the charters. For example, the Data Store charter indicates Python bindings, AWS/GCP for cloud infrastructures, Elastic Search Percolate for query service. The Data Portal charter indicates Google Analytics for system logging, AWS for hosting the website, Dockstore for hosting portals. What is the right level of including/excluding implementation choices? I think being consistent is key here across all the charters so we present a united front to our users. @brianraymor maybe you can provide some guidance? I can think of arguments both for and against mentioning JSON here.

lauraclarke replied:
I agree that implementation details should be mostly left out but I am torn on JSON as it feels like such a fundamental part of design that changing it creates significant downstream consequences and possible actually trigger rechartering 

This sounds like something we should discuss at the face to face meeting in September and have an agreement what, if any, implementation details should be in our charters

TimothyTickle replied:
I am with you Mallory, fair point. I would lean against having implementation details in any charter unless we know it would not change and it is vital to communicate the functionality at a high level. But we should come together with a common understanding.

brianraymor replied:
@malloryfreeberg - I do not believe the charters should include *internal* implementation details in their scope which should be high level. I would expect specific data formats or language bindings to be detailed in the milestones - especially since there are so many dependencies on the metadata JSON format.

Now that we have multiple charters to compare, I think that it would be helpful to focus our attention on updating Data Store as an exemplar. @kozbo has kindly agreed to withdraw his charter from architecture review this week so we can perform further review.

joshmoore replied:
In the way of an outsiders perspective, if the charter did _not_ mention the json(-schema) aspect then I would be inclined to ask about metadata specifications in other dialects. I'm guessing that would quickly take you out of scope and so I'd imagine it's for the best to have it in. Similarly to the AWS/GCP aspects brought up by @malloryfreeberg, as someone keen on `#dcp-reuse` _outside of_ the commercial clouds, unless the charter specifies that the intent is to produce software that works specifically in AWS and/or GCP, then I will eventually be inquiring about the option of OpenStack and/or Kubernetes.

lauraclarke replied:
As I said to @TimothyTickle above and I think the conclusion we reached in this conversation the JSON aspect of the schemas are so fundamental to the schemas that changing it would need rechartering. I am leaving it in.

brianraymor committed:
Updates to section headings

The charter template was updated - #18

malloryfreeberg commented:
@lauraclarke it might be worth adding an explicit statement about including documentation for helping interested parties be able to use/implement the metadata schemas. It also might just be implied. Maybe see what other Charters are doing about this.

jonahcool commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,66 @@
+* Providing exemplar data in the schema format
+* Providing computationally readable information about depreciated and moved fields within the schema
+* Working with data contributors to understand what types of descriptive fields they can supply for their samples and experiments
+* Working with data consumers to understand what descriptive fields they need in order to be able to analyse and visualise the HCA data
```
It may be out of scope but I think it is worth calling out the importance of working with contributors/consumers to understand fields but also to understand usability. This may fall here but could also be incorporated more explicitly above as a part of incorporating community feedback.

diekhans replied:
I believe working with contributors and consumers is very important and would be covered under objective 8:

"8. Ensuring the defined metadata schema provides a positive user experience for those who read and write them"

lauraclarke replied:
@jonahcool are you happy this is covered by the objective

`Ensuring the defined metadata schema provides a positive user experience for those who read and write them`

and the inscope statements

```
Working with data contributors to understand what types of descriptive fields they can supply for their samples and experiments
Working with data consumers to understand what descriptive fields they need in order to be able to analyse and visualise the HCA data
Extending the schema to meet the needs of data contributors and data consumers based on the feedback they provide
```

morrisonnorman committed:
Updated dates in Milestones and Deliverables

kbergin commented:
Hello!

Would you like to add a reference to the new upload team email on this charter?

On the Data Processing Charter I decided to make the title of the charter `mailto:email@data.humancellatlas.org` and to add it to the Communication section. I formatted it as
`[Team email](mailto:email@data.humancellatlas.org): email@data.humancellatlas.org`

For your reference, your team email is `metadata-team@data.humancellatlas.org`

Thanks! I'll be posting this on each charter for convenience, please do let me know your thoughts :)

lauraclarke committed:
Added rational about how comprehensive we aim to be

lauraclarke committed:
clarification of inscope statement

lauraclarke committed:
Clarification fix about exemplar schema

lauraclarke committed:
Added team email

lauraclarke commented:
> metadata-team



> Hello!
> 
> Would you like to add a reference to the new upload team email on this charter?
> 
> On the Data Processing Charter I decided to make the title of the charter `mailto:email@data.humancellatlas.org` and to add it to the Communication section. I formatted it as
> `[Team email](mailto:email@data.humancellatlas.org): email@data.humancellatlas.org`
> 
> For your reference, your team email is `metadata-team@data.humancellatlas.org`
> 
> Thanks! I'll be posting this on each charter for convenience, please do let me know your thoughts :)

This has been done, thanks

malloryfreeberg commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,76 @@
+3. Ensuring that HCA metadata has a consistent representation and can accurately represent the scientific description of a dataset in a computational readable manner
+4. Defining JSON schema which enable objectives 1 and 2
+5. Defining the semantic rules that the content follows
+6. Working with Ingest Infrastructure to specify the entity structure for HCA metadata schema
```
> with Ingest Infrastructure

Maybe change to "with Ingestion Service" to be consistent with how the component is referred to in other charters

lauraclarke committed:
A couple of documentation notes in the scope field

lauraclarke committed:
Fixing terminology for ingestion service

brianraymor committed:
truely -> truly

https://en.wiktionary.org/wiki/truely

lauraclarke committed:
Update to bundle requirements and responsibilities between components

lauraclarke committed:
Adding definitions and fixing spelling of deprecated

lauraclarke committed:
Added additional structure to the scope section and some edits

Particularly, I removed this scope statement

* Incorporating community feedback to extend and improve the schema

as I felt it was better expressed and therefore a duplicate of 

 Extending the schema to meet the needs of data contributors and data consumers based on the feedback they provide

I also removed the word documentation from the support statements

lauraclarke commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,95 @@
+* Providing exemplar data in JSON format that validates against the schema 
+* Providing computationally readable information about deprecated and moved fields within the schema
+
+### Schema testing
```
@malloryfreeberg @diekhans Can you review and potentially extend this section to reflect Mark's testing plan. Please do it in the next 48hrs so everyone can have enough time for a last review before community review is closed.

malloryfreeberg committed:
Added a Schema testing in-scope item

lauraclarke committed:
turn of phrase/spelling update on testing scope

brianraymor committed:
Editorial changes to Slack Channels

Adopt best practice for slack channel format
Deleted redundant markdown for mailto and https

brianraymor commented on charters/MetadataSchema/charter.md:
```diff
@@ -0,0 +1,96 @@
+
+metadata-team@data.humancellatlas.org
+
+## Github repositories
```
Can you add a description for the repo?

lauraclarke replied:
Done

lauraclarke committed:
fixed typo

lauraclarke committed:
Added description to git repo

mweiden commented:
Approved in the 2018-10-25 Tech Architecture meeting.
